pkg/utils/httpClient.go

```diff
@@ -20,8 +20,12 @@ import (
 	"golang.org/x/time/rate"
 )
 
+var (
```

Comment:
seems this is too intrusive to the http libarary.
The standard way to add user-agent into aws clients with aws-sdk-go is `https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/pkg/aws/user_agent.go#L11`

Reply:
Thanks! I was looking for something like this.